kowalski/alert_broker_pgir.py

```diff
@@ -120,6 +129,21 @@ def process_alert(alert: Mapping, topic: str):
                     upsert=True,
                 )
 
+            # Crossmatch exisiting alert with most recent record in ZTF_alerts and update aux
+            with timer(
+                f"ZTF Cross-match of {object_id} {candid}", alert_worker.verbose > 1
+            ):
+                xmatches_ztf = alert_worker.alert_filter__xmatch_live_stream(alert)
+
+            with timer(
+                f"Aux updating of {object_id} {candid}", alert_worker.verbose > 1
+            ):
+                alert_worker.mongo.db[alert_worker.collection_alerts_aux].update_one(
+                    {"_id": object_id},
+                    {"$addToSet": {"ZTF_alerts": {"$each": xmatches_ztf}}},
```

Comment:
Did you mean
```suggestion
                    {"$set": {"ZTF_alerts": xmatches_ztf}},
```
? What you have here will add the latest ZTF alert to this list (treating it as a set to avoid duplicates).

Reply:
Right, your suggestion looks better since we would avoid adding new alerts from the same ZTF name repeatedly. Okay to include only the latest ZTF alert.